parse/node.go

```diff
@@ -81,10 +82,42 @@ func (n *Node) Validate() error {
 
 		return fmt.Errorf("%s: too many keys", n.Pos())
 	}
-
+	if err := validateName(n.Name()); err != nil {
+		return fmt.Errorf("%s: %s", n.Pos(), err)
+	}
 	return n.setValues()
 }
 
+func validateName(name string) error {
+	var hasSpaces bool
+	var hasNonAlphaNum bool
+	var invalidRunes []rune
+	for _, letter := range []rune(name) {
+		if unicode.IsSpace(letter) {
+			hasSpaces = true
+		}
+		if !(unicode.IsLetter(letter) || unicode.IsNumber(letter) || letter == '-' || letter == '_' || letter == '.') {
+			hasNonAlphaNum = true
+			invalidRunes = append(invalidRunes, letter)
+		}
+	}
+	if hasSpaces {
+		return errors.New("resource name may not contain spaces")
+	}
+	if hasNonAlphaNum {
+		var badChars []string
+		printed := make(map[rune]struct{})
+		for _, runeChar := range invalidRunes {
+			if _, ok := printed[runeChar]; !ok {
+				badChars = append(badChars, string(runeChar))
+				printed[runeChar] = struct{}{}
+			}
+		}
+		return fmt.Errorf("invalid character(s) in resource name: %v", badChars)
```

Comment:
I assume this is a user-facing error message. Should it include something like "valid characters are unicode letters and numbers, dashes, underscores, and dots"?